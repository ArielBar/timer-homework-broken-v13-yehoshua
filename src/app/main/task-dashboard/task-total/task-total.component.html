<mat-card>
  <mat-card-title>
    Total: {{ totalTime | minuteSecondsFormat }}
  </mat-card-title>
</mat-card>
