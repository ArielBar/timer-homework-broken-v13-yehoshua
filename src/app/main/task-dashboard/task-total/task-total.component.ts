import { Component, ChangeDetectionStrategy, Input } from '@angular/core';

@Component({
  selector: 'app-task-total',
  templateUrl: './task-total.component.html',
  styleUrls: ['./task-total.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TaskTotalComponent {
  @Input() totalTime: number;

  constructor() {
  }
}
