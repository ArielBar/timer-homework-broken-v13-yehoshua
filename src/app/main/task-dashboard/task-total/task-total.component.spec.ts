import { ComponentFixture, TestBed } from '@angular/core/testing';
import { TaskTotalComponent } from './task-total.component';
import { MatCardModule } from '@angular/material/card';
import { Pipe, PipeTransform } from "@angular/core";

@Pipe({ name: 'minuteSecondsFormat' })
class MockPipe implements PipeTransform {
  transform(value: number): number {
    return value;
  }
}

describe('TaskContainerComponent', () => {
  let component: TaskTotalComponent;
  let fixture: ComponentFixture<TaskTotalComponent>;

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [TaskTotalComponent, MockPipe],
      imports: [MatCardModule],
    });
    fixture = TestBed.createComponent(TaskTotalComponent);
    component = fixture.componentInstance;
  });

  it('can load instance', () => {
    expect(component).toBeTruthy();
  });

  describe('DOM', () => {
    it('should render totalTime$', () => {
      component.totalTime = 20;
      fixture.detectChanges();

      fixture.whenStable().then(() => {
        expect(fixture.nativeElement).toMatchSnapshot();
      });
    });
  });
});
