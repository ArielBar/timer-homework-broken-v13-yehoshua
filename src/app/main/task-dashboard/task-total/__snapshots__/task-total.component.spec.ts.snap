// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskContainerComponent DOM should render totalTime$ 1`] = `
<div
  id="root1"
>
  <mat-card
    class="mat-card mat-focus-indicator"
  >
    <mat-card-title
      class="mat-card-title"
    >
       Total: 20 
    </mat-card-title>
  </mat-card>
</div>
`;
